src/os/bluestore/BlueStore.cc

```diff
@@ -8335,9 +8352,20 @@ void BlueStore::_txc_release_alloc(TransContext *txc)
     // bstore_kv_final. We could serialize in simpler format and perform
     // the merge separately, maybe even in a dedicated thread.
     bulk_release_extents.insert(txc->released);
+    int r = 0;
+    if (cct->_conf->bdev_enable_discard && cct->_conf->bdev_async_discard) {
+      r = bdev->queue_discard(bulk_release_extents);
+      if (r == 0)
+	goto out;
```

Comment:
@voidbag I think we need to invoke alloc->release() even though bdev->queue_discard is fail?